### ISSUE TITLE
User `.ncurc` is ignored when `--global` flag is used

### ISSUE THREAD

**Issue by duniul** [raineorshine/npm-check-updates#1299]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 14.14`

---

## Steps to Reproduce

.ncurc.json:

```json
{
  "packageManager": "pnpm"
}
```

Dependencies (pnpm global):

```json
"dependencies": {
	"npm-check-updates": "16.10.10",
	"prettier": "2.8.8",
	"typescript": "^5.0.4"
}
```

Steps:
- Add a root `.ncurc` with any settings
- Run `ncu` without `-g` (works as expected, config is used)
- Run `ncu -g` (config is not loaded)

## Current Behavior

The user `.ncurc` is not used when the `-g/--global` flag is used, only CLI args are respected.

Passing `--configFilePath` or `--configFileName` does not help.

In my case, it's blocking me from using `"packageManager": "pnpm"` for global deps by default.

## Expected Behavior

The user `.ncurc` should be loaded even if the `-g/--global` flag is used.

## Reason

Looking at the code in `src/index.ts`, it looks like the global check is happening before the config is loaded, so it only gets passed the CLI options (https://github.com/raineorshine/npm-check-updates/blob/main/src/index.ts#L166-L179).

## Alternative

If this is intentional to avoid mixing up the global options with project configs, then would it be possible to add a way to define a separate config for global settings?

Fantastic tool btw 🙌

**Comment by raineorshine** [comment]
Hi, thanks for the suggestion. npm-check-updates should definitely be able to read a config file when in global mode.

I have added automatic detection of an ncurc file in the home directory (`/Users/NAME` on Linux/OSX and `C:\Users\NAME` on Windows). I have also added support for `--configFilePath` to be used with `--global`.

Published in `16.10.12`

**Comment by duniul** [comment]
<img src="https://media4.giphy.com/media/yXVO50FJIJMSQ/giphy.gif"/>

So fast! v16.10.12 works just as expected, thanks! 🙏
